src/Raven.Server/Web/System/AdminDumpHandler.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Properties;
+using Raven.Server.Rachis;
+using Raven.Server.Routing;
+using Raven.Server.Utils;
+using Sparrow.Platform;
+using Sparrow.Utils;
+
+namespace Raven.Server.Web.System
+{
+    public class AdminDumpHandler : ServerRequestHandler
+    {
+        [RavenAction("/admin/debug/dump", "GET", AuthorizationStatus.ClusterAdmin)]
+        public async Task Dump()
+        {
+            var typeAsString = GetStringQueryString("type");
+            if (Enum.TryParse(typeAsString, out DumpType type) == false)
+                throw new InvalidOperationException($"Could not parse '{typeAsString}' to dump type.");
+
+            var path = GetStringQueryString("path", required: false);
+            if (string.IsNullOrWhiteSpace(path))
+                path = Path.GetTempPath();
+
+            using (var process = Process.GetCurrentProcess())
+            {
+                string fileName = GetFileName(".dmp");
+                path = Path.Combine(path, fileName);
+                HttpContext.Response.RegisterForDispose(new DeleteFile(path));
+
+                Execute($"dump --type {type}", process.Id, path);
+
+                using (var file = File.OpenRead(path))
+                {
```

Comment:
We *have* to do compression here.
It may make the change from 100GB to 8 GB sizes.

Reply:
Done